### PR TITLE
Render payment form with errors

### PR DESCRIPTION
**Pull request by davidrapson**
It was bugging me that we don't handle form errors server-side. This PR updates the checkout form to render-with errors if there is an issue.

**Warning** I've just Play documentation copy-pasted and IntelliJ autocompleted my way to a solution for this. I'm sure there is a more idiomatic way to do this; but it was a useful learning experiment trying this.

I'm also aware that Play has a bunch of form helpers so I'm probably repeating some things, but as we're not using them anywhere it felt OK to follow suit and keep the error class handling manual.

screen shot 2015-08-14 at 14 43 26

@afiore

### CODE REVIEW

File: app/controllers/Checkout.scala [hunk 1]
```diff
@@ -1,46 +1,41 @@
...
 import utils.TestUsers.{NameEnteredInForm, PreSigninTestCookie}
 import views.html.{checkout => view}
-
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-import AuthenticationService.authenticatedUserFor
 
 object Checkout extends Controller with LazyLogging {
 
+  def fillForm(form: Form[SubscriptionData], authUserOpt: Option[AuthenticatedIdUser]): Future[Form[SubscriptionData]] = {
```
Comment (afiore) [inline app/controllers/Checkout.scala]: I agree with moving this one out of `renderCheckout`, but I would also mark it as private

**Comment by afiore** [comment]
:+1: LGTM

File: app/views/fragments/checkout/fieldsPayment.scala.html [hunk 2]
```diff
@@ -1,11 +1,15 @@
 @(form: Form[model.SubscriptionData])
 
-<div class="form-field js-checkout-account">
+@hasError(key: String) = @{
```
Comment (afiore) [inline app/views/fragments/checkout/fieldsPayment.scala.html]: If you find yourself repeating this kind of code too much, remember that you can always create helper methods under the `views` package, and they will be available in the view without the need of explicitly importing them

**Comment by davidrapson** [comment]
@TesterSpike Look OK to you?

**Pull request closed without merging**